testing/scenario_app/ios/Scenarios/ScenariosUITests/AppExtensionTests.m

```diff
@@ -27,13 +32,25 @@ - (void)testAppExtensionLaunching {
   }
   [button tap];
   BOOL launchedExtensionInFlutter = NO;
-  // Custom share extension button (like the one in this test) does not have a unique
-  // identity. They are all identified as `XCElementSnapshotPrivilegedValuePlaceholder`.
-  // Loop through all the `XCElementSnapshotPrivilegedValuePlaceholder` and find the Flutter one.
+
+  // Wait for first cell of share sheet to appear.
+  XCUIElement* firstCell = self.hostApplication.collectionViews.cells.firstMatch;
+  if (![firstCell waitForExistenceWithTimeout:10]) {
+    NSLog(@"%@", self.hostApplication.debugDescription);
+    XCTFail(@"Failed due to not able to find any cells with %@ seconds", @(10));
+  }
+
+  // Custom share extension button (like the one in this test) does not have a
+  // unique identity on older versions of iOS. They are all identified as
+  // `XCElementSnapshotPrivilegedValuePlaceholder`. On iOS 17, they are
+  // identified by name. Loop through all the buttons labeled
+  // `XCElementSnapshotPrivilegedValuePlaceholder` or `Scenarios` to find the
+  // Flutter one.
   for (int i = 0; i < self.hostApplication.collectionViews.cells.count; i++) {
```

Comment:
I know you didn't add this code but can you make it a `for`/`in` instead?  You can also use a predicate (I didn't actually run this so it may not be quite right):
```objc
  NSPredicate* cellPredicate =
      [NSPredicate predicateWithFormat:@"label == 'XCElementSnapshotPrivilegedValuePlaceholder' OR label = 'Scenarios'"];
  NSArray<XCUIElement*>* shareSheetCells = [self.hostApplication.collectionViews.cells matchingPredicate:cellPredicate].allElementsBoundByIndex;
  for (XCUIElement* shareSheetCell in shareSheetCells) {
```